protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -301,6 +364,11 @@ func (rpccs *RPCConsumerServer) SendRelay(
 		utils.LavaFormatDebug("relay succeeded after retries", utils.Attribute{Key: "GUID", Value: ctx}, utils.Attribute{Key: "retries", Value: retries})
 	}
 	rpccs.appendHeadersToRelayResult(ctx, returnedResult, retries)
+
+	if RelaysHealthEnable {
+		rpccs.relaysMonitor.LogRelay()
```

Comment:
instead of having this if case everywhere in the code, just add a nil check into the relayMonitor 
```go
func (rm *RelayMonitor) blabla() {
   if rm == nil {
       return 
   }
}
```

Reply:
and make sure its nil when not RelaysHealthEnable = false...

Reply:
Fixed in 0d17caa346eb14e794b6c77db0b92a0a533c75d1